Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -350,7 +356,22 @@ class CombinatorialKalmanFilter {
                            << tipState.nMeasurements
                            << ", nOutliers = " << tipState.nOutliers
                            << ", nHoles = " << tipState.nHoles << " on track");
-              result.trackTips.emplace_back(currentTip);
+              result.lastTrackIndices.emplace_back(currentTip);
+              // Set the lastMeasurementIndex to the last measurement
+              // to ignore the states after it in the rest of the algorithm
+              auto lastMeasurementIndex = currentTip;
+              bool isMeasurement =
+                  result.fittedStates.getTrackState(lastMeasurementIndex)
+                      .typeFlags()
+                      .test(TrackStateFlag::MeasurementFlag);
+              while (!isMeasurement) {
+                auto lastMeasurementState =
+                    result.fittedStates.getTrackState(lastMeasurementIndex);
+                lastMeasurementIndex = lastMeasurementState.previous();
+                isMeasurement = lastMeasurementState.typeFlags().test(
+                    TrackStateFlag::MeasurementFlag);
+              }
```

Comment:
So just for my understanding, this just walks back the trajectory and modifies the `lastMeasurementIndex` until it finds a measurement, and records that value into `lastMeasurementIndices`, correct?

Reply:
Yes that all it does.